src/Sarif.Converters/ContrastSecurityConverter.cs

```diff
@@ -0,0 +1,1364 @@
+                {
+                    return ConstructEventValidationDisabledResult(context.Properties);
+                }
+
+                case ContrastSecurityRuleIds.FormsAuthenticationSSL:
+                {
+                    return ConstructFormsAuthenticationSSLResult(context.Properties);
+                }
+
+                case ContrastSecurityRuleIds.FormsWithoutAutocompletePrevention:
+                {
+                    return ConstructFormsWithoutAutocompletePreventionResult(context.Properties);
+                }
+
+                case ContrastSecurityRuleIds.HttpOnlyCookieFlagDisabled:
+                {
+                    return ConstructHttpOnlyCookieFlagDisabledResult(context.Properties);
+                }
+
+                case ContrastSecurityRuleIds.InsecureEncryptionAlgorithms:
+                {
+                    return ConstructInsecureEncryptionAlgorithmsResult(context.Properties);
+                }
+
+                case ContrastSecurityRuleIds.OverlyLongSessionTimeout:
+                {
+                    return ConstructOverlyLongSessionTimeoutResult(context.Properties);
+                }
+
+                case ContrastSecurityRuleIds.PagesWithoutAntiClickjackingControls:
+                {
+                    return ConstructPagesWithoutAntiClickjackingControlsResult(context.Properties);
+                }
+
+                case ContrastSecurityRuleIds.PathTraversal:
+                {
+                    return ConstructPathTraversalResult(context.Properties);
+                }
+
+                case ContrastSecurityRuleIds.RequestValidationModeDisabled:
+                {
+                    return ConstructRequestValidationModeDisabledResult(context.Properties);
+                }
+
+                case ContrastSecurityRuleIds.SessionCookieHasNoSecureFlag:
+                {
+                    return ConstructSessionCookieHasNoSecureFlagResult(context.Properties);
+                }
+
+                case ContrastSecurityRuleIds.SessionRewriting:
+                {
+                    return ConstructSessionRewritingResult(context.Properties);
+                }
+
+                case ContrastSecurityRuleIds.SqlInjection:
+                {
+                    return ConstructSqlInjectionResult(context);
+                }
+
+                case ContrastSecurityRuleIds.VersionHeaderEnabled:
+                {
+                    return ConstructVersionHeaderEnabledResult(context.Properties);
+                }
+
+                case ContrastSecurityRuleIds.WebApplicationDeployedinDebugMode:
+                {
+                    return ConstructWebApplicationDeployedinDebugModeResult(context.Properties);
+                }
+
+                default:
+                {
+                    return ConstructNotImplementedRuleResult(context.RuleId);
+                }
+            }
+
+            throw new InvalidOperationException();
```

Comment:
>InvalidOperationException [](start = 22, length = 25)

IMO this should be `ArgumentException`. The method was called with a `context` object containing an unrecognized `RuleId`. `InvalidOperationException` means "the object can't do what you asked in its current state".